### DISCUSSION TITLE
Need Help

### DISCUSSION THREAD

**Discussion by neophyte2** [l-lin/angular-datatables#1553] [Q&A]
am having trouble adding a button to the table ..
 Giving the bellow error :
core.js:6210 ERROR TypeError: Cannot read property 'createEmbeddedView' of undefined
    at ViewContainerRef.createEmbeddedView (core.js:23127)
    at angular-datatables.directive.js:78
    at Array.forEach (<anonymous>)
    at jQuery.fn.init.rowCallback (angular-datatables.directive.js:68)
    at jquery.dataTables.js:6682
    at Function.map (jquery.js:467)
    at _fnCallbackFire (jquery.dataTables.js:6681)
    at _fnDraw (jquery.dataTables.js:3503)
    at _fnAjaxUpdateDraw (jquery.dataTables.js:4169)
    at jquery.dataTables.js:4009

**Comment by shanmukhateja** [comment]
Hi,

Please use Discussions section for support. Please provide StackBlitz link or GitHub repo to investigate this.

I'm closing this issue.

**Reply by neophyte2** [reply]
have you pulled the repo

**Reply by shanmukhateja** [reply]
@neophyte2  It asked for a password. Could you make a small reproducible repo instead?

**Reply by neophyte2** [reply]
when running the code  or pulling the repo

**Reply by shanmukhateja** [reply]
when cloning the repo. Also you seem to have removed the clone link here? 😕

**Reply by neophyte2** [reply]
yeah it's a private project
so i found a way but there is no way i can put an if stattement
{
          title: 'Action',
          render: function (data: any, type: any, full: any) {
            return '<div id="dropdownMenuLink2" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" style="font-weight: bold;">' +
              ' <i class="fa fa-ellipsis-h"></i>' +
              '</div>' +
              '<div class="dropdown-menu" aria-labelledby="dropdownMenuLink2">' +
              ' <a class="dropdown-item" view="' + full.clientID + '">View' +
              '</a>' +
              ' <a class="dropdown-item" enable="' + full.fullname + '" enable2="' + full.ID + '" enable3="' + full.status + '">Enable</a>' +
              '<a class="dropdown-item" disable="' + full.fullname + '" disable2="' + full.ID + '" disable3="' + full.status + '">Disable</a>' +
              ' </div> '
          }
        },
